lib/application.js

```diff
@@ -398,85 +399,217 @@ app.enableAuth = function(options) {
     }
   };
 
-  this._verifyAuthModelRelations();
-
-  this.isAuthEnabled = true;
+  return this._verifyAuthModelRelations(app._warnOnBadAuthModelRelations)
+    .then(() => {
+      this.isAuthEnabled = true;
+    });
```

Comment:
as `_verifyAuthModelRelations` is now async and returns a promise, this change is required
also i made enableAuth() return a promise to, as it's de facto async too now, it's not a breaking change to me, the method can still be called in an app while not caring about the returned promise if most if not all use cases.

This is useful in the tests so we can wait for enableAuth to return before checking the  displayed warnings